great_expectations/expectations/registry.py

```diff
@@ -89,10 +90,38 @@ def register_expectation(expectation: Type["Expectation"]) -> None:  # noqa: F82
             logger.warning(
                 f"Overwriting declaration of expectation {expectation_type}."
             )
+
     logger.debug(f"Registering expectation: {expectation_type}")
     _registered_expectations[expectation_type] = expectation
 
 
+def register_data_assistant(
+    data_assistant: Type["DataAssistant"],  # noqa: F821
+) -> None:
+    data_assistant_type = data_assistant.data_assistant_type
+    if data_assistant_type in _registered_data_assistants:
+        if _registered_data_assistants[data_assistant_type] == data_assistant:
+            logger.info(
+                f'Multiple declarations of DataAssistant "{data_assistant_type}" found.'
+            )
+            return
+        else:
+            logger.warning(
+                f'Overwriting the declaration of DataAssistant "{data_assistant_type}" took place.'
+            )
+
+    logger.debug(
+        f'Registering the declaration of DataAssistant "{data_assistant_type}" took place.'
+    )
+    _registered_data_assistants[data_assistant_type] = data_assistant
+
+
+def get_data_assistant_impl(
+    data_assistant_type: str,
+) -> Type["DataAssistant"]:  # noqa: F821
+    return _registered_data_assistants.get(data_assistant_type)
```

Comment:
Is this nullable? If so, we should make the return `Optional`. If not, we should use `[]` to make it very explicit that we're confident we have the value in the dict.

Reply:
@cdkini Fixed.  Thanks.